apps/passport-client/src/dispatch.ts

```diff
@@ -842,6 +842,10 @@ async function doSync(
         serverStorageHash: upRes.value.storageHash
       };
     } else {
+      if (upRes.error.name === "ValidationError") {
+        return { userInvalid: true };
```

Comment:
Excellent point in your commit comment about why this doesn't cause a loop.
Can you capture that point in a comment for the next person to examine this code?